crates/webidl/Cargo.toml

```diff
@@ -20,7 +20,7 @@ heck = "0.3"
 log = "0.4.1"
 proc-macro2 = "1.0"
 quote = '1.0'
-syn = { version = '2.0', features = ['full'] }
+syn = { version = '2.0', features = ['full', 'extra-traits'] }
```

Comment:
Note to self: figure out how to remove this requirement.

Reply:
I took a quick look at it, one approach could be to do the check on the witter type here: https://github.com/rustwasm/wasm-bindgen/blob/a9f06701c78c9623772fe3f1a3e2cf0d60241114/crates/webidl/src/lib.rs#L380 and then pass along a boolean `is_js_value_ref_option_type`.

Reply:
Please go ahead and do that then.

Reply:
I found a way to do it by matching the idl type and replicating the logic from IdlType::to_syn_type but I don't love it.. it feels much less robust. It's your call though.

Reply:
This is pretty ugly indeed, but I think it's very important for a library like `wasm-bindgen` to reduce its dependency footprint.